scipy/optimize/_trustregion_constr/minimize_trustregion_constr.py

```diff
@@ -439,7 +441,7 @@ def stop_criteria(state, x, last_iteration_failed,
                     return True
             if state.optimality < gtol and state.constr_violation < gtol:
                 state.status = 1
-            elif state.tr_radius < xtol:
+            elif state.tr_radius < xtol and state.constr_violation < gtol:
```

Comment:
By adding this extra stopping condition the docstring for `xtol` becomes incorrect.

The docstring should read something like:

> Tolerance for termination by the change of the independent variable.
        The algorithm will terminate when ``tr_radius < xtol and constr_violation < gtol``, where
        ``tr_radius`` is the radius of the trust region used in the algorithm.
        Default is 1e-8.


In any case, I'm not sure if this is the desired behaviour. I'd prefer to keep the stopping condition unchanged (i.e. revert this line). Instead should we add an extra item to `TERMINATION_MESSAGES`, "constraints are not satisfied", and change L538 to:

```python
if state.constr_violation > gtol:
    result.status = 4    # TERMINATION_MESSAGES[4] = "constraints are not satisfied"
result.success = True if result.status in (1, 2) else False
```

My rationale is that it's no change to the stopping condition, but still informs the user that something has gone wrong. It's relatively common that the constraints are always infeasible, in which case the proposed PR would keep going until maximum iterations had been reached. That's undesirable as it artificially prolongs minimisation.

I don't think adding the extra condition in L442 would actually improve a minimisation anyway. I'm ready to be convinced with evidence though.